core-api/src/main/java/com/redhat/lightblue/BaseResponse.java

```diff
@@ -0,0 +1,236 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.util.Error;
+import com.redhat.lightblue.util.JsonObject;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Common response information from mediator APIs
+ */
+public class BaseResponse extends JsonObject {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String PROPERTY_ENTITY = "entity";
+    private static final String PROPERTY_VERSION = "entityVersion";
+    private static final String PROPERTY_STATUS = "status";
+    private static final String PROPERTY_MOD_COUNT = "modifiedCount";
+    private static final String PROPERTY_MATCH_COUNT = "matchCount";
+    private static final String PROPERTY_TASK_HANDLE = "taskHandle";
+    private static final String PROPERTY_SESSION = "session";
+    private static final String PROPERTY_DATA_ERRORS = "dataErrors";
+    private static final String PROPERTY_ERRORS = "errors";
+    private static final String PROPERTY_HOSTNAME = "hostname";
+
+    private EntityVersion entity;
+    private OperationStatus status;
+    private long modifiedCount;
+    private long matchCount;
+    private String taskHandle;
+    private SessionInfo session;
+    private String hostname;
+    private final List<DataError> dataErrors;
+    private final List<Error> errors;
+
+    final JsonNodeFactory jsonNodeFactory;
+
+    private static final String HOSTNAME;
+
+    static {
+        String hostName = "unknown";
+        try {
+            InetAddress localHost = InetAddress.getLocalHost();
+            if (localHost != null) {
+                hostName = localHost.getHostName();
+            }
+        } catch (UnknownHostException e) {
+
+        }
+        HOSTNAME = hostName;
+    }
+
+    /**
+     * @deprecated use Response(JsonNodeFactory)
+     */
+    @Deprecated
+    public BaseResponse() {
```

Comment:
Pushed down from Response. I moved everything in Response except entityData and entityMetadata to BaseResponse, and this moved as well.